pkgs/by-name/sh/sharkey/package.nix

```diff
@@ -0,0 +1,158 @@
+{ lib
+, stdenv
+, stdenvNoCC
+, fetchFromGitLab
+, makeWrapper
+, copyDesktopItems
+, jemalloc
+, ffmpeg-headless
+, jq
+, python3
+, pkg-config
+, glib
+, vips
+, moreutils
+, cacert
+, nodePackages
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sharkey";
+  version = "2024.3.1";
+
+  src = fetchFromGitLab {
+    owner = "TransFem-org";
+    repo = "Sharkey";
+    domain = "activitypub.software";
+    rev = "${finalAttrs.version}";
+    hash = "sha256-+lu0l/TA2Ge/flTUyyV/i0uzh4aycSGVCSQMkush8zA=";
+    fetchSubmodules = true;
+  };
+
+  # NOTE: This requires pnpm 8.10.0 or newer
+  # https://github.com/pnpm/pnpm/pull/7214
+  pnpmDeps =
+    assert lib.versionAtLeast nodePackages.pnpm.version "8.10.0";
+    stdenvNoCC.mkDerivation {
+      pname = "${finalAttrs.pname}-pnpm-deps";
+      inherit (finalAttrs) src version;
+
+      nativeBuildInputs = [
+        jq
+        moreutils
+        nodePackages.pnpm
+        cacert
+      ];
+
+      pnpmPatch = builtins.toJSON {
+        pnpm.supportedArchitectures = {
+          os = [ "linux" ];
+          cpu = [ "x64" "arm64" ];
+        };
+      };
+
+      postPatch = ''
+        mv package.json package.json.orig
+        jq --raw-output ". * $pnpmPatch" package.json.orig > package.json
+      '';
```

Comment:
ime running pnpm with `--force`(as in #290715) seems to be more reliable than this package.json patching in some cases involving cross-platform builds? not sure on the conditions of such a thing happening, so take with a pinch of salt

Reply:
`--force` is better, yes. See https://github.com/pnpm/pnpm/issues/7695

But I am still not very confident on this fixed-output derivation as it does not provide stable hashes across different versions of pnpm

Reply:
hmm, how can we execute scripts without making it unreproducible?